lib/mongodb.js

```diff
@@ -275,6 +310,9 @@ MongoDB.prototype.create = function(model, data, options, callback) {
     data._id = oid; // Set it to _id
     idName !== '_id' && delete data[idName];
   }
+
+  data = self.toDatabase(model, data);
```

Comment:
I find it a bit suspicious, why it was not necessary to call `toDatabase` before? @superkhau @raymondfeng you are more familiar with this code base, do you know the answer?